plugins/modules/network_settings_workflow_manager.py

```diff
@@ -2321,7 +2321,7 @@ def get_want_reserve_pool(self, reserve_pool):
                     self.status = "failed"
                     return self
 
-                if not pool_values.get("ipv4PrefixLength"):
+                if not pool_values.get("ipv4PrefixLength") and pool_values.get("ipv4Prefix") is True:
```

Comment:
By placing the ipv4Prefix check first, you ensure ipv4PrefixLength is only evaluated if ipv4Prefix exists and is True, removing the need to compare it explicitly to True.

if pool_values.get("ipv4Prefix") and not pool_values.get("ipv4PrefixLength"):
    self.msg = "Missing parameter 'ipv4_prefix_length' in reserve_pool_details '{0}' element" \
               .format(reserve_pool_index + 1)
    self.status = "failed"

Reply:
simplified the check.